src/store.js

```diff
@@ -26,24 +26,51 @@ export default new Vuex.Store({
         description: "Very nice cup"
       }
     ],
-    cart: [{
+    cart: {
       orderId: 1,
       status: 1,
-      products: [
-
-      ]
-    }],
+      count: 0,
+      products: []
+    },
     taxonomies: {
 
     }
   },
   mutations: {
     TOGGLE_MENU(state) {
-      state.menu.visible = !state.menu.visible;
+      state.menu.visible = !state.menu.visible
     },
     ADD_PRODUCT_TO_STORAGE(state, product) {
-      state.nextProductId++;
+      state.nextProductId++
       state.allProducts.push(product)
+    },
+    REMOVE_PRODUCT_FROM_CART(state, productId) {
+      state.cart.products.forEach((item, index) => {
+        if (item.productId == productId) {
+          state.cart.products.splice(index, 1)
+          state.cart.count--;
+        }
+      })
+    },
+    ADD_PRODUCT_TO_CART(state, {
+      productId,
+      quantity
+    }) {
+      let itemExist = false;
+      state.cart.products.forEach((item, index) => {
```

Comment:
Jeżeli produkt o danym ID występuje w tablicy tylko raz, to lepiej użyć zwykłej pętli `for` i po znalezieniu productu wywołać `break`, żeby nie przechodzisz nie potrzebnie po całej tablicy

Reply:
done.